framework/sandstone_ssl.h

```diff
@@ -11,17 +11,33 @@
 #include <openssl/bio.h>
 #include <openssl/blowfish.h>
 #include <openssl/cmac.h>
+#include <openssl/core_names.h>
 #include <openssl/crypto.h>
 #include <openssl/evp.h>
 #include <openssl/hmac.h>
 #include <openssl/md5.h>
+#include <openssl/modes.h>
 //#include <openssl/mdc2.h>
 //#include <openssl/ripemd.h>
 #include <openssl/pem.h>
+#include <openssl/provider.h>
+#include <openssl/rand.h>
 #include <openssl/rsa.h>
 #include <openssl/sha.h>
 //#include <openssl/whrlpool.h>
 
+/* Define macros based on OpenSSL defined macros
+ * - openssl/params.h
+ * TODO: Find a better way to fix this */
+# define s_OSSL_PARAM_DEFN(key, type, addr, sz)    \
```

Comment:
We can use macros directly, without `s_`.

Reply:
I guess we can, as long as we re-defined the macros to call the prefixed functions or we are ending up linking  against OpenSSL on all cases.